staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -146,3 +151,33 @@ type CustomResourceValidation struct {
 	// OpenAPIV3Schema is the OpenAPI v3 schema to be validated against.
 	OpenAPIV3Schema *JSONSchemaProps
 }
+
+// CustomResourceSubResources defines the status and scale subresources for CustomResources.
+type CustomResourceSubResources struct {
+	// Status denotes the status subresource for CustomResources
+	Status *CustomResourceSubResourceStatus
+	// Scale denotes the scale subresource for CustomResources
+	Scale *CustomResourceSubResourceScale
+}
+
+// CustomResourceSubResourceStatus defines how to serve the HTTP path <CR Name>/status.
+// status is represented by the `.status` JSON path.
+type CustomResourceSubResourceStatus struct{}
+
+// CustomResourceSubResourceScale defines how to serve the HTTP path <CR name>/scale.
+type CustomResourceSubResourceScale struct {
+	// required, e.g. “.spec.replicas”.
+	// Only JSON paths without the array notation are allowed.
+	SpecReplicasPath string
+	// optional, e.g. “.status.replicas”.
```

Comment:
why is `StatusReplicas` optional?  If it defaults to something otherwise, you should probably write that here (same with the other optional fields)

Reply:
If I remember right, the idea was that a scale subresource could exist without the scale being reflected in the status. So there is no defaulting of this field.

Reply:
Adding on to what @sttts said, the user cannot write to the `/status` at all. It is completely read-only for the user. Only a controller has write access to the status fields. So we can leave it empty and not have it default to anything. This also matches our plan with CRDs.